lib/collection/variable.js

```diff
@@ -131,10 +131,8 @@ _.assign(Variable.prototype, /** @lends Variable.prototype */ {
         }
 
         // set the new type if it is valid and cast the stored value
-        if (typeName) {
-            this.type = typeName;
-            !_nocast && (this.value = this.cast(this.value));
-        }
```

Comment:
There is an early return for empty `typeName`. By this line, `typeName` should always be present right?.